inloop/solutions/management/commands/generate_submissions.py

```diff
@@ -0,0 +1,117 @@
+
+    def create_users(self, amount):
+        with file_dir.joinpath("animals.txt").open(mode="r") as f:
+            usernames = [animal.strip().lower() for animal in f.readlines()]
+        chosen_names = random.sample(usernames, amount)
+        users = []
+        for name in chosen_names:
+            username = "Anonymous {}".format(name)
+            email = "{}@example.com".format(name)
+            password = "secret"
+            # Since username is a unique constraint, it needs to be
+            # checked first.
+            if User.objects.filter(username=username).exists():
+                users.append(User.objects.get(username=username))
+            else:
+                users.append(User.objects.create_user(
+                    username=username, email=email, password=password
+                ))
+        return users
+
+    def create_solution(self, user, task, amount, start_date, end_date):
+        # Randomly choose, if the solution succeeded
+        passed = random.choice([True, False])
+        # Generate random submission date between start and end date
+        submission_date = start_date + timedelta(
+            seconds=random.randint(0, int((end_date - start_date).total_seconds()))
+        )
+        solution = Solution.objects.create(
+            author=user, task=task, passed=passed,
+        )
+        # If auto_now_add is active, the submission date must be supplied
+        # after object creation.
+        solution.submission_date = submission_date
+        solution.save()
+        return solution
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--start_date",
+            help="Start date (format {})".format(self.dateformat),
+            default=(datetime.now() - timedelta(days=365)).strftime(self.dateformat),
+        )
+        parser.add_argument(
+            "--end_date",
+            help="(Optional) end date (format {})".format(self.dateformat),
+            default=datetime.now().strftime(self.dateformat),
+        )
+        parser.add_argument(
+            "--solutions_amount",
+            help="Number of solutions to generate",
+            type=int,
+            default=1000
+        )
+        parser.add_argument(
+            "--users_amount",
+            help="Number of users to generate",
+            type=int,
+            default=10
+        )
+        parser.add_argument(
+            "--force",
+            help="Immediately perform submission creation without sanity checks",
+            default=False
+        )
+        parser.add_argument(
+            "--silent",
+            help="Do not print to console",
+            default=False
+        )
+
+    def handle(self, *args, **options):
+        start_date = make_aware(datetime.strptime(options["start_date"], self.dateformat))
+        end_date = make_aware(datetime.strptime(options["end_date"], self.dateformat))
+        solutions_amount = options["solutions_amount"]
```

Comment:
Still some `amounts` lying around (ten instances, to be precise). `grep -i amount generate_submissions.py` ;-)